domains/certificates/RegistryConnector/Worker/EventHandlers/MeasurementEventHandler.cs

```diff
@@ -110,33 +85,30 @@ public async Task Consume(ConsumeContext<EnergyMeasuredIntegrationEvent> context
                 message.Quantity,
                 commitment.BlindingValue.ToArray());
 
-            dbContext.Add(consumptionCertificate);
-
-            var (ownerPublicKey, issuerKey) = keyGenerator.GenerateKeyInfo(message.Quantity,
-                matchingContract.WalletPublicKey, walletDepositEndpointPosition.Value, matchingContract.GridArea);
-
-            var issuedEvent = Registry.CreateIssuedEventForConsumption(
-                projectOriginRegistryOptions.RegistryName,
-                consumptionCertificate.Id,
-                period.ToDateInterval(),
-                matchingContract.GridArea,
-                message.GSRN,
-                commitment,
-                ownerPublicKey);
-
-            var transaction = issuedEvent.CreateTransaction(issuerKey);
+            logger.LogInformation("Creating consumption certificate for {Message} with {certificateId}", message, consumptionCertificate.Id);
 
-            BuildRoutingSlip(builder, transaction, commitment, matchingContract, walletDepositEndpointPosition.Value);
+            dbContext.Add(consumptionCertificate);
 
-            logger.LogInformation("Created consumption certificate for {Message}", message);
+            certificateId = consumptionCertificate.Id;
+            logger.LogInformation("Created consumption certificate for {certificateId}", message);
         }
         else
             throw new CertificateDomainException(string.Format("Unsupported metering point type {0} for message {1}",
                 matchingContract.MeteringPointType, message));
 
-        var routingSlip = builder.Build();
-
-        await context.Execute(routingSlip);
+        await context.Publish<CertificateCreatedEvent>(new CertificateCreatedEvent
+        {
+            MeteringPointType = matchingContract.MeteringPointType,
+            Period = period,
+            Quantity = quantity,
+            CertificateId = certificateId,
+            WalletPublicKey = matchingContract.WalletPublicKey,
+            WalletUrl = matchingContract.WalletUrl,
+            GridArea = matchingContract.GridArea,
+            Gsrn = gsrn,
+            WalletDepositEndpointPosition = walletDepositEndpointPosition.Value,
+            Technology = technology
```

Comment:
Couldnt you use
`Technology = technology ?? null `
instead of setting null at line 79?

Reply:
I like that it is explicitly set to null tho, for readability